### ISSUE TITLE
Review: Not showing files ending +ZZ (a timezone)

### ISSUE THREAD

**Issue by wxtim** [cylc/cylc-flow#4260]
**Describe the bug**
@MetRonnie  found the following bug whilst reviewing something else.

> However, when I try to access a specific log file (one with a datetime in the filename), I get a 404 page with this traceback, both for Cylc 7 & 8 workflows:
>```
> Traceback (most recent call last):
>   File "~/github/cylc-7/lib/cherrypy/_cprequest.py", line 679, in respond
>     response.body = self.handler()
>   File "~/github/cylc-7/lib/cherrypy/lib/encoding.py", line 230, in __call__
>     self.body = self.oldhandler(*args, **kwargs)
>   File "~/github/cylc-7/lib/cherrypy/_cpdispatch.py", line 60, in __call__
>     return self.callable(*self.args, **self.kwargs)
>   File "~/github/cylc-7/lib/cylc/review.py", line 666, in view
>     user, suite, path, path_in_tar=path_in_tar, mode=mode)
>   File "~/github/cylc-7/lib/cylc/review.py", line 505, in get_file
>     f_name = self._get_user_suite_dir(user, suite, path)
>   File "~/github/cylc-7/lib/cylc/review.py", line 930, in _get_user_suite_dir
>     return self._check_dir_access(path)
>   File "~/github/cylc-7/lib/cylc/review.py", line 789, in _check_dir_access
>     raise cherrypy.HTTPError(404)
> HTTPError: (404, None)
> ```
> Seeing as it happens for both 7/8, I guess this is a separate issue

I have confirmed that this is still an issue. Further investigation adding 
```diff
# lib/cylc/review.py@line772
- raise cherrypy.HTTPError(404)
+ raise cherrypy.HTTPError(404, 'File Not Found: {}'.format(path))
```
Shows that the `+` between the time and timezone has been replaced by a space. Depending on time constraints this problem.

**Release version(s) and/or repository branch(es) affected?**
7.8.x

**Steps to reproduce the bug**
- With Cylc 7.8.x or 8 run a workflow.
- With Cylc@7.8.x run `cylc review start`.
- Open `localhost:8080` in a browser and navigate to a workflow
- Enter either the cycles or task job listing pages then goto the cylc files dropdown (top rightish)
- Select a file in the form `??/??/log.YYYYMMDDTHHMMSS+ZZ`

**Expected behavior**
You should be shown a Cylc Log file.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by MetRonnie** [comment]
This is closed by #4276, right?
